server/src/main/java/org/elasticsearch/rest/RestRequest.java

```diff
@@ -108,6 +110,7 @@ private RestRequest(NamedXContentRegistry xContentRegistry, Map<String, String>
         this.rawPath = path;
         this.headers = Collections.unmodifiableMap(headers);
         this.requestId = requestId;
+        this.compatibleVersion = CompatibleVersionHelper.getCompatibleVersion(parsedAccept, parsedContentType, hasContent());
```

Comment:
it is tempting to have that method private to RestRequest, but at the same time it is fairly big and would cause RestRequest to do even more. Testing is also easier. I recon we keep this as a static helper.